demos/text_detection_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,349 @@
+            // Text Recognition
+            cv::GArray<cv::GMat> texts;
+            if (trComposite) {
+                cv::GArray<cv::GMat> hiddens;
+                cv::GArray<cv::GMat> features;
+                std::tie(hiddens, features) =
+                    cv::gapi::infer2<nets::TextRecognitionEncoding>(inRec, labels);
+                texts = custom::CompositeTRDecode::on(hiddens, features,
+                                                      config.decoderNumClasses,
+                                                      config.decoderEndToken);
+            } else {
+                texts = cv::gapi::infer2<nets::TextRecognition>(inRec, labels);
+            }
+            outs += cv::GOut(texts);
+        } else {
+            pts = custom::PointsFromRRects::on(rrs, size, FLAGS_cc);
+        }
+        outs += cv::GOut(pts);
+        // Inputs and outputs of graph
+        cv::GComputation graph(cv::GIn(in), std::move(outs));
+        /** ---------------- End of graph ---------------- **/
+
+        // Getting information about frame from ImagesCapture
+        std::shared_ptr<ImagesCapture> cap = openImagesCapture(FLAGS_i, FLAGS_loop, read_type::safe);
+        const cv::Mat tmp = cap->read();
+        cap.reset();
+        cap = openImagesCapture(FLAGS_i, FLAGS_loop, read_type::safe);
+
+        auto compileArgs = cv::compile_args(custom::kernels(), config.networks);
+        if (trComposite) {
+            compileArgs.emplace_back(CompositeDecInputDescrs{
+                                        cv::GMatDesc(CV_32F, config.decoderHiddenInputDims),
+                                        cv::GMatDesc(CV_32F, config.decoderFeaturesInputDims)
+                                     });
+        }
+        cv::GStreamingCompiled pipeline;
+        if (gapiStreaming) {
+            pipeline = graph.compileStreaming(std::move(compileArgs));
+            pipeline.setSource(cv::gin(cv::gapi::wip::make_src<custom::CommonCapSrc>(cap)));
+            pipeline.start();
+        }
+
+        LazyVideoWriter videoWriter{FLAGS_o, cap->fps(), FLAGS_limit};
+
+        cv::Size graphSize{static_cast<int>(tmp.cols / 4), 60};
+        Presenter presenter(FLAGS_u, tmp.rows - graphSize.height - 10, graphSize);
+
+        // Output containers for results
+        cv::Mat image;
+        std::vector<std::vector<cv::Point2f>> outPts;
+        std::vector<cv::Mat> outTexts;
+        auto getOutVector = [&](){
+            auto outVector = gapiStreaming ? cv::gout(image) : cv::gout();
+            if (trEnabled) {
+                outVector += cv::gout(outTexts);
+            }
+            outVector += cv::gout(outPts);
+            return outVector;
+        };
+
+        /** ---------------- The execution part ---------------- **/
+        PerformanceMetrics metrics;
+        std::chrono::steady_clock::time_point beginFrame = std::chrono::steady_clock::now();
+        while (true) {
```

Comment:
```suggestion
        while (true) {
            std::chrono::steady_clock::time_point beginFrame = std::chrono::steady_clock::now();
```
With that you can remove `beginFrame = std::chrono::steady_clock::now();` in the end